platforms/android/app/src/main/java/com/maq/xprize/cci/hindi/SplashScreenActivity.java

```diff
@@ -0,0 +1,147 @@
+                } else {
+                    Toast.makeText(this, "Permission required!", Toast.LENGTH_LONG).show();
+                    finish();
+                }
+            }
+            break;
+            case 2: {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    Intent intent = new Intent(this, MainActivity.class);
+                    startActivity(intent);
+                } else {
+                    Toast.makeText(this, "Permission required!", Toast.LENGTH_LONG).show();
+                    finish();
+                }
+            }
+            break;
+        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        // call main activity if extraction done
+        String flagFilePath = "/storage/emulated/0/Android/data/com.maq.xprize.CCI.hindi/files/.success.txt";
+        File flagFile = new File(flagFilePath);
+        if (flagFile.exists()) {
+            if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                    != PackageManager.PERMISSION_GRANTED ) {
+                // Permission is not granted
+                ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, 2);
+            } else {
+                Intent intent = new Intent(this, MainActivity.class);
+                startActivity(intent);
+                finish();
+            }
+        }
+
+        super.onCreate(savedInstanceState);
+        this.requestWindowFeature(Window.FEATURE_NO_TITLE);
+        this.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
+        if (Build.VERSION.SDK_INT < 19) { // lower api
+            View v = this.getWindow().getDecorView();
+            v.setSystemUiVisibility(View.GONE);
+        } else {
+            //for new api versions.
+            View decorView = this.getWindow().getDecorView();
+            int uiOptions = View.SYSTEM_UI_FLAG_HIDE_NAVIGATION | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY;
+            decorView.setSystemUiVisibility(uiOptions);
+        }
+        setContentView(R.layout.activity_splash_screen);
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                != PackageManager.PERMISSION_GRANTED ||
+                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION)
+                        != PackageManager.PERMISSION_GRANTED) {
+            // Permission is not granted
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE,
+                    Manifest.permission.ACCESS_FINE_LOCATION}, 1);
+        } else {
+            new DownloadFile().execute(null, null, null);

```

Comment:
Add the logic to check whether the expansion file needs to be extracted even if the success.txt is present.
You can refer to onebillion's PR

Reply:
Will require update and testing after adding the logic. Kept in the backlog for now